scripts/retrieveTranslations.js

```diff
@@ -0,0 +1,116 @@
+const fs = require('fs');
+const util = require('util');
+const {CimpressTranslationsClient} = require("cimpress-translations");
+const KmsAuthenticator = require("./kmsClientIdAuthorizer");
+const iso639 = require('iso-639').iso_639_2;
+
+const SERVICE_ID = '60eff5ed-9ddf-483e-b60a-3135c3c87435';
+const CLIENT_ID = 'C9twFCynWeNXTu74hxGMzlId745Togdb';
+const ENCRYPTED_CLIENT_SECRET = 'AQECAHgPvC4p26eHImEv+Cm3nU1nSoO1Mca9Hldw/zbnVdYjtQAAAKIwgZ8GCSqGSIb3DQEHBqCBkTCBjgIBADCBiAYJKoZIhvcNAQcBMB4GCWCGSAFlAwQBLjARBAxNPmCBj+vSauoVQxUCARCAWwA74kzSlZn7psPFtAS/yn5AzqvxdPzuiHzlMWXAI8yheQC2NDPacwQgmyst0zeViWEjSzhXmsRXA47O1qTux92Go15G6vnvEbVJvEGS0KVOQWNhD8Iv5shlYrE=';
+const TRANSLATIONS_PATH = __dirname + '/../src/locales';
+
+async function retrieveTranslations() {
+
+    // Decrypt
+    log('Retrieving Auth0 token...');
+    let authenticator = new KmsAuthenticator(CLIENT_ID, ENCRYPTED_CLIENT_SECRET);
+    let token = await authenticator.getAccessToken();
+
+    // Init translation client
+    log(`Retrieving service ${SERVICE_ID} description.`);
+    let translationClient = new CimpressTranslationsClient(null, `Bearer ${token}`);
+    [err, description] = await to(translationClient.describeService(SERVICE_ID));
+    if ( err ) return 1;
+    log('Description:');
+    log(util.inspect(description, false, null));
+
+    log(`Service description retrieved. Found languages: ${JSON.stringify(description.languages.map(s => s.blobId))}`);
+    let languagePromises = [];
+    description.languages.forEach(lang => {
+        log(`Retrieving language translation for ${lang.blobId}`);
+        languagePromises.push(translationClient.getLanguageBlob(SERVICE_ID, lang.blobId));
+    });
+
+    [err, langs] = await to(Promise.all(languagePromises));
+    if ( err ) return 1;
+    log(`Done.`);
+
+    if ( !fs.existsSync(TRANSLATIONS_PATH) ) {
+        log('Creating translations folder...');
+        fs.mkdirSync(TRANSLATIONS_PATH);
+    }
+
+    let allJS_imports = '';
+    let allJS_translations = '';
+    let twoLetterLang = {};
+    let allJS_react_timeago_imports = '';
+    let allJS_react_timeago_builders = '';
+
+    langs.forEach(blob => {
+        let translationFile1 = `${TRANSLATIONS_PATH}/translations.${blob.blobId}.json`;
+        log(`Storing translation ${translationFile1}`);
+        fs.writeFileSync(translationFile1, JSON.stringify(blob.data, null, 4));
+
+        twoLetterLang[blob.blobId] = iso639[blob.blobId]["639-1"] || blob.blobId;
+        allJS_translations += `    ${blob.blobId}: ${blob.blobId},\n`;
+        allJS_imports += `import ${blob.blobId} from './translations.${blob.blobId}.json';\n`;
+        allJS_react_timeago_imports += `import ${blob.blobId}Strings from 'react-timeago/lib/language-strings/${twoLetterLang[blob.blobId]}';\n`;
+
+        // react-timeago builders
+        allJS_react_timeago_builders += `    ${blob.blobId}: buildFormatter(${blob.blobId}Strings),\n`
+    });
+
+    let allJS = `
```

Comment:
I know this looks a bit odd ;) It generates a .js file that includes all the localization data for the current package + all needed information to translate react-timeago (which requires 2-digit country code and preloaded translation files as well).